### PR TITLE
Add protoc-go-inject-tag to nixpkgs

### PR DESCRIPTION
**Pull request by ElrohirGT**
## Description of changes

This is a new package used for go-tags and protobuffers: [Homepage](https://github.com/favadi/protoc-go-inject-tag)

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/pr/protoc-go-inject-tag/package.nix [hunk 1]
```diff
@@ -0,0 +1,27 @@
+{
+  lib,
+  buildGoModule,
+  fetchFromGitHub,
+  ...
```
[the remark this answers is not in the document]
Comment (ElrohirGT) [reply pkgs/by-name/pr/protoc-go-inject-tag/package.nix:5]: Makes sense, I'll remove it

**Comment by lolbinarycat** [comment]
all the files look good, one last step, you need to squash the last two commits together so you have:
1. one commit adding yourself to `maintainers`
2. a second commit adding the file

i believe `git rebase -i` is the reccomended method for this.

**Comment by ElrohirGT** [comment]
Nice! I don't have my PC with me right now, I'll do it later

**Comment by ElrohirGT** [comment]
Done!

**Comment by ElrohirGT** [comment]
It builds on M1 too!

**Comment by RaghavSood** [comment]
@ElrohirGT are you able to clear the merge conflicts? PR looks good otherwise

**Comment by ElrohirGT** [comment]
Sure! I'll clean them up this afternoon. It should still only contain two commits right? The one that adds me as maintainer and the one that adds the package.

**Comment by RaghavSood** [comment]
Yes, should still be only two - you should be able to replay the commits over the current master and then fix the conflict in the maintainers file - rebase and squash as before if needed

**Comment by ElrohirGT** [comment]
Sorry! I think I messed up a little and closed this PR on accident while updating my fork. I don't really know how that happened probably by force pushing. Anyways, here's the [fixed PR](https://github.com/NixOS/nixpkgs/pull/302624).

**Pull request closed without merging**